### ISSUE TITLE
lfe_io:format issue with binary

### ISSUE THREAD

**Issue by oubiwann** [lfe/lfe#123]
This issue was noted on the mail list late last year:
- https://groups.google.com/d/msg/lisp-flavoured-erlang/pF4retTadzw/M_8jsuM2vkMJ

Here's the set up:

``` cl
> (set encoded (binary ("åäö ð" utf8)))
#"åäö ð"
```

Here's `io:format` working as expected:

``` cl
> (io:format "~tp~n" `(,encoded))
<<"åäö ð"/utf8>>
ok
```

Here's `lfe_io` making the same call:

``` cl
> (lfe_io:format "~tp~n" `(,encoded))
exception error: badarg
  in (: lfe_io fwrite1 "~tp~n" (#"åäö ð"))
  in lfe_io:format/3 (src/lfe_io.erl, line 179)
```

**Comment by rvirding** [comment]
The error is because `lfe_io:format` doesn't support the `t` modifier. Now it works like this for `io`:

```
> (set encoded (binary ("åäö ð" utf8)))
#"åäö ð"
> (io:format "~w\n" `(,encoded))    
<<195,165,195,164,195,182,32,195,176>>
ok
> (io:format "~p\n" `(,encoded))
<<"Ã¥Ã¤Ã¶ Ã°">>
ok
> (io:format "~tp\n" `(,encoded))   
<<"åäö ð"/utf8>>
ok
```

The `~w` just writes out the raw bytes in the binary. Without the `t` modifier `~p` will print the binary as a sequence of latin1 characters and with the modifier as a utf-8 encoded binary.

My plan for LFE (and `lfe_io:format`) is that a `~w` will write out the raw bytes and if the binary can be a utf-8 encoded string then `~p` will be print it as a utf-8 binary string else the raw bytes:

```
> (lfe_io:format "~w\n" `(,encoded))
#B(195 165 195 164 195 182 32 195 176)
ok
> (lfe_io:format "~p\n" `(,encoded))
#"åäö ð"
ok
```

The rationale behind this is that if I have a syntax for inputting utf-8 encoded binary then this should be the default when printing them. The same as for list strings.

The LFE repl uses `lfe_io:format` for output.

**Comment by rvirding** [comment]
This was fixed in commit d9b4057a9bd0f0240c5bb150ed9a977a7cb14a95 so LFE now supports #"..." as syntax for utf-8 encode binaries and it will prettyprint utf-8 encoded binaries like this.
